experiment/simulation/js/gate.js

```diff
@@ -0,0 +1,317 @@
+import { registerGate, jsPlumbInstance } from "./main.js";
+import { setPosition } from "./layout.js";
+import { decoderTest, displayTest, computeAnd, computeOr, computeXor, computeXnor, computeNand, computeNor } from "./validator.js";
+
+'use strict';
+
+export let gates = {}; // Array of gates
+window.numComponents = 0;
+export function clearGates() {
+
+    for (let gateId in gates) {
+        delete gates[gateId];

```

Comment:
Deleting from a collection while iterating over it is generally not a good idea. Why do you need to call delete function here when you are already clearing the whole object below. If you are trying to have some side-effects, there may be a better way to achieve the side-effects without deleting the element e.g. the deleteElement function at the bottom of this file.

Reply:
Yes, i just wanted to clear the object. Dont need delete for that I guess.